### PR TITLE
Add proxy in devServer to bypass the CORS issue

### PR DESCRIPTION
**Pull request by JordyMoos**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,35 +46,15 @@ $ cd yapm-server
 $ composer install
 ```
 
-Super cheaty CORS fix in `index.php`
-```php
-// Add to top of the file
-// This allows cross origin requests
-header('Access-Control-Allow-Origin: *');
-header('Access-Control-Allow-Methods: POST, GET, OPTIONS');
-header('Access-Control-Allow-Credentials: true');
```
Comment (JordyMoos) [inline README.md]: This part is not needed anymore

File: README.md [hunk 2]
```diff
@@ -46,35 +46,15 @@ $ cd yapm-server
...
-
-// Add just before $app->run();
-// This catches all "options" methods
-$app->match("{url}", function($url) use ($app) { return "OK"; })->assert('url', '.*')->method("OPTIONS");
```
Comment (JordyMoos) [inline README.md]: This is already in the yapm-server code

File: config.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "apiEndPoint": "http://localhost:8001",
+  "apiEndPoint": "http://localhost:3000/server",
```
Comment (JordyMoos) [inline config.json]: The default server entpoint is the same port as the client, but with a path prefix /server (kinda how i setup mine in production).

File: webpack.config.js [hunk 4]
```diff
@@ -179,7 +179,15 @@ if (TARGET_ENV === 'development') {
         devServer: {
             inline: true,
             stats: 'errors-only',
-            contentBase: path.join(__dirname, "src/static/assets")
+            contentBase: path.join(__dirname, "src/static/assets"),
+            proxy: {
+                '/server': {
```
Comment (JordyMoos) [inline webpack.config.js]: And what happends in the devServer is that all calls to /server are intercepted, and rerouted like configured below the comment

File: README.md (hunk 1 above)
Comment (marcusklaas) [reply README.md]: 👍

File: webpack.config.js (hunk 4 above)
Comment (marcusklaas) [reply webpack.config.js]: Clever!

**Review by marcusklaas — APPROVED** [review]
This is a great way to clean up the CORS hack! 💯

**Pull request merged**